### PR TITLE
Add resource product

### PR DESCRIPTION
**Pull request by gabrielgiroe1**
Fix: https://github.com/avo-hq/avo/issues/2786

### CODE REVIEW

File: test/models/product_test.rb
Comment (adrianthedev) [inline test/models/product_test.rb:1]: let's remove this if we won't write tests for it.

File: test/factories/products.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FactoryBot.define do
+  factory :product do
+    title { "MyString" }
+    description { "MyText" }
+    status { "MyString" }
+    category { "MyString" }
+    price_cents { 1 }
+    price_currency { "MyString" }
+    created_at { "2024-05-23 10:34:41" }
+    updated_at { "2024-05-23 10:34:41" }
```
Comment (adrianthedev) [inline test/factories/products.rb:10]: Same. let's remove it if we won't use it.

**Review by adrianthedev — CHANGES_REQUESTED** [review]
Looking good, but let's add some seed data as well, please.

File: test/factories/products.rb (hunk 1 above)
Comment (gabrielgiroe1) [reply test/factories/products.rb:10]: ok

**Review by adrianthedev — CHANGES_REQUESTED** [review]
let's add some images as well please.
I think the dummy app has some set up

**Pull request merged**